apps/andi/lib/andi/input_schemas/ingestion.ex

```diff
@@ -63,7 +63,7 @@ defmodule Andi.InputSchemas.Ingestion do
     source_format = Map.get(changes, :sourceFormat, nil)
 
     ingestion
-    |> cast(changes_with_id, @cast_fields, empty_values: [])
+    |> cast(changes_with_id, @cast_fields, empty_values: [""])
```

Comment:
Fixes a bug in product that prevents removing a dataset. (Appears to remove, but if you save and return, it restores the previous dataset)

Explanation: Because dataset is a required foreign key, the changeset rejects "" (empty string) as a valid change and falls back to the existing data. However, the changeset allows nil to be a valid change, effectively unsetting the field.

This line adds "" as an empty value, converting it to nil, therefore allowing the metadata form to unset the changeset field.